newIDE/app/src/GameDashboard/GameCard.js

```diff
@@ -1,92 +1,331 @@
 import * as React from 'react';
+import { type I18n as I18nType } from '@lingui/core';
+
+import { Card, Chip } from '@material-ui/core';
+import ShareIcon from '@material-ui/icons/Share';
+import MoreVert from '@material-ui/icons/MoreVert';
+
 import { Column, Line, Spacer } from '../UI/Grid';
 import RaisedButton from '../UI/RaisedButton';
-import { getGameUrl, type Game } from '../Utils/GDevelopServices/Game';
 import { ResponsiveLineStackLayout } from '../UI/Layout';
-import Window from '../Utils/Window';
+import FlatButton from '../UI/FlatButton';
+import IconButton from '../UI/IconButton';
+import Text from '../UI/Text';
+import Toggle from '../UI/Toggle';
+import CardContent from '../UI/Card/CardContent';
+import CardHeader from '../UI/Card/CardHeader';
+import ElementWithMenu from '../UI/Menu/ElementWithMenu';
+
 import { GameThumbnail } from './GameThumbnail';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import ShareDialog from './ShareDialog';
+
+import {
+  getGameUrl,
+  updateGame,
+  type Game,
+} from '../Utils/GDevelopServices/Game';
+import Window from '../Utils/Window';
+import { type GamesDetailsTab } from './GameDetailsDialog';
+import Dialog from '../UI/Dialog';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import { showErrorBox } from '../UI/Messages/MessageBox';
 
 type Props = {|
   game: Game,
   isCurrentGame: boolean,
-  onOpenGameManager: () => void,
+  onOpenGameManager: (tab: GamesDetailsTab) => void,
+  onUpdateGame: () => Promise<void>,
 |};
 
-export const GameCard = ({ game, isCurrentGame, onOpenGameManager }: Props) => {
+type TogglableProperties =
+  | 'discoverable'
+  | 'acceptsBuildComments'
+  | 'acceptsGameComments';
+
+const confirmationMessage = {
+  discoverable: {
+    true: <Trans>Are you sure you want to make this game discoverable?</Trans>,
+    false: (
+      <Trans>
+        Are you sure you don't want to make this game discoverable anymore?
+      </Trans>
+    ),
+  },
+  acceptsBuildComments: {
+    true: (
+      <Trans>
+        Are you sure you want to ask for feedbacks on all build pages?
+      </Trans>
+    ),
+    false: (
+      <Trans>
+        Are you sure you want to stop asking for feedbacks on all build pages?
+      </Trans>
+    ),
+  },
+  acceptsGameComments: {
+    true: (
+      <Trans>
+        Are you sure you want to show a feedback banner on Liluo.io?
+      </Trans>
+    ),
+    false: (
+      <Trans>
+        Are you sure you want to remove the feedback banner on Liluo.io?
+      </Trans>
+    ),
+  },
+};
+
+export const GameCard = ({
+  game,
+  isCurrentGame,
+  onOpenGameManager,
+  onUpdateGame,
+}: Props) => {
   const openGameUrl = () => {
     const url = getGameUrl(game);
     if (!url) return;
     Window.openExternalURL(url);
   };
+  const [showShareDialog, setShowShareDialog] = React.useState(false);
+  const [
+    showConfirmationDialog,
```

Comment:
I'd rename this prop, I got confused and thought it was a boolean deciding if the dialog is opened or not.
Maybe `shownConfirmationDialog`/`setShownConfirmationDialog` (switch focus from "show" to "dialog")

Or maybe better: `confirmationDialogName` ?